core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -153,14 +163,31 @@ class AdminClient(val time: Time,
                                   consumers: Option[List[ConsumerSummary]],
                                   coordinator: Node)
 
-  def describeConsumerGroup(groupId: String): ConsumerGroupSummary = {
-    val coordinator = findCoordinator(groupId)
+  def describeConsumerGroupHandler(coordinator: Node, groupId: String): GroupMetadata = {
     val responseBody = send(coordinator, ApiKeys.DESCRIBE_GROUPS,
         new DescribeGroupsRequest.Builder(Collections.singletonList(groupId)))
     val response = responseBody.asInstanceOf[DescribeGroupsResponse]
     val metadata = response.groups.get(groupId)
     if (metadata == null)
       throw new KafkaException(s"Response from broker contained no metadata for group $groupId")
+    metadata
+  }
+
+  def describeConsumerGroup(groupId: String): ConsumerGroupSummary = describeConsumerGroup(groupId, 0)
+
+  def describeConsumerGroup(groupId: String, timeoutMs: Long): ConsumerGroupSummary = {
+    timerMs = time.milliseconds
+
+    val coordinator = findCoordinator(groupId, timeoutMs)
+    var metadata = describeConsumerGroupHandler(coordinator, groupId)
+
+    while (metadata.state != "Dead" && metadata.state != "Empty" &&
```

Comment:
Actually, I think this will change the logic a bit. The current implementation keeps trying the API call without throwing an exception, but in the suggested snippet it would throw the exception on the first failed try. Initially I missed this, and noticed that some of the relevant unit tests are intermittently failing.

Plus, in the suggested snippet we are not checking metadata `state` and `protocolType` after the last API call (in the loop).

Please advise if I'm mistaken.